### PR TITLE
91 interpolation

### PR DESCRIPTION
**Pull request by emercado4**
Merge request template: please remove the appropriate parts of this template. 

Pre-merge request checklist (to be completed by the one making the request):
 - [x] I have performed a full review of this code myself.
    - For Python code in PySpark specific sections, all code should have been run in Jupyter notebooks. 
    - For code in sections of the book containing both Python and R code, the page of the book should be constructed as described in the [contributing guide](https://github.com/best-practice-and-impact/ons-spark/blob/main/CONTRIBUTING.md) and converted to a markdown file. 
 - [x] I have formatted the outputs of code blocks correctly (to match other outputs in the book and in line with the style guide [coming soon])
 - [x] I have built the book as outlined in the [contributing guide](https://github.com/best-practice-and-impact/ons-spark/blob/main/CONTRIBUTING.md) and confirmed that any additional/modified content is displaying as expected.

Details of this request:
 - Adds [interpolation.ipynb tip](https://gitlab-app-l-01/DAP_CATS/troubleshooting/python-troubleshooting/-/blob/master/interpolation.ipynb?ref_type=heads) to analysis in spark section of the book
 - Equivalent sparklyr code has also been added in. 

Requirements for review (such as):
 - [x] Check that the plot images display correctly
 - [x] Check that the code tabs display correctly
 - [x] Review text for readability, typos etc
 - [x] Check that links all work correctly
 - [x] Check that code works as expected

### CODE REVIEW

File: ons-spark/spark-analysis/interpolation.md
Comment (Jday7879) [inline ons-spark/spark-analysis/interpolation.md:367]: Missing "{code-tab} r R" When converting to R script this block was missed causing an error, with this corrected the script runs correctly

**Review by Jday7879 — CHANGES_REQUESTED** [review]
Book builds with figures and equations displaying correctly. A few small issues with the text and some links, but doesn't need many changes. Missing markdown code `{code-tab} r R` caused a section to not copy over when converting back, when I copied this both python and R scrips ran without error.
Nice page and content layout :)

File: ons-spark/spark-analysis/interpolation.md
Comment (emercado4) [reply ons-spark/spark-analysis/interpolation.md]: I _think_ that mixed frequency points is referring to different time gaps between consecutive periods. I've added some additional explanation now to try to clarify.

File: ons-spark/spark-analysis/interpolation.md
Comment (emercado4) [reply ons-spark/spark-analysis/interpolation.md]: I agree. I've reworded this now, hopefully its a bit easier to read?

File: ons-spark/spark-analysis/interpolation.md
Comment (emercado4) [reply ons-spark/spark-analysis/interpolation.md:507]: Thanks! I've added a link to the window functions page and to the visualisation page now. I'm not sure if there are any others that would be good to include? There might be some more to add eventually too as the Analysis in Spark section continues to grow.

File: ons-spark/spark-analysis/interpolation.md
Comment (Jday7879) [reply ons-spark/spark-analysis/interpolation.md]: Extra info does clarify things now (in my opinion!)

File: ons-spark/spark-analysis/interpolation.md
Comment (Jday7879) [reply ons-spark/spark-analysis/interpolation.md]: I couldn't quite tell what needed to change the first time but this is much easier to read now :)

File: ons-spark/spark-analysis/interpolation.md
Comment (Jday7879) [reply ons-spark/spark-analysis/interpolation.md:507]: After Checking with Nathan,  the URLs for book pages needs to be defined using the relative path '../spark-functions/window-functions' etc. This is just in case the book URL changes at any point in the future. Everything else looks good to merge once update :)

**Review by Jday7879 — CHANGES_REQUESTED** [review]
Just one final change needs to be made to spark book links before merging, I agree with all other changes or comments

**Review by Jday7879 — APPROVED** [review]
Book build correctly, ready to merge :)

**Pull request merged**
